File: 151.py
class Solution:
    def reverseWords(self, s: str) -> str:
        s = s.strip()
        firstPointer = len(s)-1
        lastPointer = len(s)-1
        
        output = ''
        while (firstPointer >= 0):
    

            if s[firstPointer] != ' ' and (s[firstPointer-1] == ' ' or firstPointer == 0):
                output += s[firstPointer:lastPointer+1] + ' '
                firstPointer -= 2
                lastPointer = firstPointer
            elif  s[firstPointer] == ' ':
                firstPointer -= 1
                lastPointer -= 1
            else:
                firstPointer -= 1

        output = output.strip() 
        return output
    
solution = Solution()
s = '   hello world   asdf '
solution.reverseWords(s)

File: 151.java
class Solution {
    public static void main(String[] args){
        String s = "hello world";
        System.out.println(reverseWords(s));
    }
    public static String reverseWords(String s) {
        s = s.trim();
        int firstPointer = s.length()-1;
        int lastPointer = s.length()-1;

        String output = "";
        while (firstPointer >= 0){
            System.out.println(s.substring(firstPointer,firstPointer+1));
            if (! s.substring(firstPointer,firstPointer+1).equals(" ") && (firstPointer ==0 || s.substring(firstPointer-1,firstPointer).equals(" "))){
                output += s.substring(firstPointer,lastPointer+1) + " ";
                
                firstPointer -= 2;
                lastPointer = firstPointer;
            } else if (s.substring(firstPointer,firstPointer+1).equals(" ")){
                firstPointer -= 1;
                lastPointer -= 1;
            } else {
                firstPointer -= 1;
            }
        }
        output = output.trim();
        return output;
    }
}